### PR TITLE
[flutter_tools] Throw in flutter test with device

### LINKED ISSUE #125070: Running `flutter test -d chrome` silently runs non-web tests.
## Description

When I run:

```script
flutter test --verbose -d chrome
```

I can see in the output that it is running the host (Linux in this case) version of the unit test, not the web version, but it just happily runs the test and reports the result.

`flutter test` shouldn't accept the `-d` argument if it is going to ignore it, at least in this case. It should either run in chrome, or not allow the argument.

To run the test the way I wanted, I needed to supply this instead:

```script
flutter test --verbose --platform=chrome 
```

### PR DESCRIPTION
**Pull request by christopherfujino**
Fixes https://github.com/flutter/flutter/issues/125070

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 1]
```diff
@@ -268,6 +268,9 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
 
   @override
   Future<FlutterCommandResult> verifyThenRunCommand(String? commandPath) {
+    if (stringArg('device-id', global: true) != null) {
+      throwToolExit('The --device-id (-d) option is not supported with the "test" sub-command.');
```
Comment (andrewkolos) [inline packages/flutter_tools/lib/src/commands/test.dart:272]: Nit: should we ask the user if they meant to use --platform instead?
```suggestion
      throwToolExit('The --device-id (-d) option is not supported with the "test" sub-command. Did you mean to use --platform instead?');
```

**Review by andrewkolos — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/lib/src/commands/test.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/commands/test.dart:272]: Given that we've marked `--platform` as deprecated, I don't think we should nudge users toward it.

**Comment by christopherfujino** [comment]
In looking at the failures and the code, I realize this is the wrong change to make, as the `--device-id` flag can still be parsed when running integration tests. it appears it's specifically the `--device-id=chrome` that is getting ignored.

**Pull request closed without merging**